tests/testthat/test-modify.R

```diff
@@ -77,6 +77,25 @@ test_that("can modify non-vector lists", {
   expect_equal(modify2(notlist(1), list(3, 4), ~ .y), notlist(3, 4))
 })
 
+test_that("modifying data frame ignores [<- methods", {
+  df <- function(...) {
+    structure(
+      data.frame(..., stringsAsFactors = FALSE),
```

Comment:
```suggestion
      data_frame(...),
```

You have access to the vctrs version too, if you want it